eng/versioning/find_invalid_versions.py

```diff
@@ -0,0 +1,48 @@
+import re
+from os import path
+
+from version_shared import get_packages, get_version_py
+
+SDK_PATH = "../../sdk/"
```

Comment:
Is the goal to plug this into the analyze step in CI? To catch invalid versions?

Reply:
If we did that I'd want to clean this up and provide more details in the error messages beyond today's equivalent of "this package doesn't pass one of the several checks" ... The goal here was to assist in going through the packages and finding inconsistencies that were the result of versioning information not being processed properly. I've kept it in here because it will help with that upgrading process... 

However, not all packages in the repo fit this versioning standard (mostly because they're auto generated). The auto generator is going to be updated to fix these in the near future. Having this tool fail a build today because some auto generated packages aren't using the correct standard would be bad so it'd be more of an FYI at this point. 

I'll put in some comments about how this tool can be used and a non-failing version can be integrated into some kind of nightly all-up run today.

Reply:
In my experience if we don't enforce a particular pattern then people will most likely not follow it over time. I'd suggest 2 things one is we should add a note about the versioning convention in Contributing.md and 2 we should file a work-item to clean-up this check and make it part of the analyze step.

Reply:
This is now added into the Analyze step... At this point it's using the `--always-succeed` flag so that it doesn't block the build (otherwise it would block CI for core). I have another issue open to include all of the proper package exclusion using the `.toml` file in the same folder as `setup.py`